sdks/python/apache_beam/transforms/environments.py

```diff
@@ -257,8 +311,8 @@ def __init__(
         (self.container_image))
 
   def __eq__(self, other):
-    return self.__class__ == other.__class__ \
-           and self.container_image == other.container_image
+    return (
+        super().__eq__(other) and self.container_image == other.container_image)
 
   def __hash__(self):
     return hash((self.__class__, self.container_image))
```

Comment:
This one doesn't handle resource hints. (Maybe that's OK 'cause it'll just be a collision.)

Reply:
Done